lua/glance/list.lua

```diff
@@ -352,17 +352,36 @@ function List:get_current_item()
   return item
 end
 
+---@param opts { start: integer, step_increment: integer, cycle?: boolean }
+function List:walk(opts)
+  local idx = opts.start
+  return function()
+    idx = idx + opts.step_increment
+    if opts.cycle then
+      idx = ((idx - 1) % vim.api.nvim_buf_line_count(self.bufnr)) + 1
```

Comment:
The - 1 and + 1 juggling here is to account for 0-based modulo math with 1-based Lua table indices.